ion-schema/src/main/kotlin/com/amazon/ionschema/model/AstSchema.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazon.ionschema.model
+
+import com.amazon.ionelement.api.IonElement
+
+/**
+ * Represents a complete Ion Schema document. Note—the model does not include a particular ISL version because the AST
+ * is intended to be a version-agnostic representation of ISL.
+ */
+data class AstSchema(
+    val id: String? = null,
+    val header: AstHeader,
+    val content: List<SchemaContent>,
+    val footer: AstFooter,
+) {
+
+    val types get() = content.filterIsInstance<SchemaContent.NamedType>()
+
+    sealed class SchemaContent {
+        /**
+         * Represents a top-level type declaration in a schema document.
+         */
+        data class NamedType(val name: String, val definition: AstType.TypeDefinition) : SchemaContent() {
+            @JvmOverloads constructor(
+                name: String,
+                constraints: Collection<AstConstraint<*>>,
+                userContent: List<Pair<String, IonElement>> = emptyList()
```

Comment:
What is this modeling? Is it intended to be an Ion Struct but defined a different way?

Reply:
I'm assuming you are referring specifically to this line:
```kotlin
userContent: List<Pair<String, IonElement>> = emptyList()
```

This is modeling any "open content" in the type definition. Ion Schema allows you to throw extra fields into a type definition, and Ion Schema implementations will accept it but not use it for anything. However, it's recommended that implementations provide some way for users to access the open content in the schema.

I should probably change it to `List<Field>`. Does that sound reasonable?